docs/how-to-guides/upgrade-openebs.md

```diff
@@ -0,0 +1,252 @@
+
+* A Kubernetes cluster accessible via `kubectl`.
+
+* [OpenEBS component](../../configuration-reference/components/openebs-operator)
+  installed. You can check if OpenEBS is indeed in expected version:
+    ```bash
+    $ kubectl get pods -n openebs -l openebs.io/version=1.12.0
+    ```
+
+* Upgrade process should not be disruptive, but it is recommended to schedule a
+  downtime for the applications consuming the OpenEBS PV and make sure to take a
+  backup of the data before starting the below upgrade procedure in case some
+  problem arises. Lokomotive provides
+  [Velero](../../configuration-reference/components/velero) component for backup
+  and restore.
+
+* Ensure the cluster and OpenEBS volumes are is in healthy state before proceeding.
+    ```bash
+    $ lokoctl health
+    Node                    Ready    Reason          Message
+
+    alpha-controller-0      True     KubeletReady    kubelet is posting ready status
+    alpha-large-worker-0    True     KubeletReady    kubelet is posting ready status
+    Name      Status    Message              Error
+
+    etcd-0    True      {"health":"true"}
+
+
+   $ kubectl get cstorpools -n openebs # Status should be Healthy.
+
+   NAME                                ALLOCATED   FREE   CAPACITY   STATUS    READONLY   TYPE      AGE
+   cstor-pool-openebs-replica-1-w3r7   6.98G       437G   444G       Healthy   false      striped   4d22h
+
+
+   $ kubectl get cstorvolume -n openebs # Status should be Healthy.
+
+   NAME                                       STATUS    AGE     CAPACITY
+   pvc-183dc3df-a7d9-4273-a8e2-7f66d2e19f4e   Healthy   3d20h   50Gi
+   pvc-5d4b4c2b-2ed5-4e75-aeb9-fbd2918ebb78   Healthy   3d20h   50Gi
+   ```
+## Steps
+
+Set the following environment variables in the terminal to assist in the upgrade process:
+```bash
+export OPENEBS_OLD_VERSION=1.12.0
+export OPENEBS_NEW_VERSION=2.2.0
+```
+### Step 1: Upgrade OpenEBS control plane components
+
+Lokomotive provides an easy way of upgrading the OpenEBS control plane.
+
+Execute the following command to upgrade OpenEBS control plane:
+
+```bash
+lokoctl component apply openebs-operator
+```
+
+Doing so, terminates the OpenEBS resources associated with the old version and new
+resources are created for the new version.
+
+Verify all the pods are in `Running` state before proceeding:
+
+```bash
+$ kubectl get pods -n openebs -l openebs.io/version=${OPENEBS_NEW_VERSION}
+```
+
+### Step 2: Upgrade OpenEBS data plane components
+
+OpenEBS control plane and data plane components work independently. Even if the
+control plane components are upgraded to the new version, the data plane components
+continue to work with the older version.
+
+#### Upgrade cStor pools
+
+Create a Job to upgrade the existing cStor pools:
+
+```bash
+cat > upgrade-cstor-pools-from-${OPENEBS_OLD_VERSION}-to-${OPENEBS_NEW_VERSION}.yaml <<EOF
+# upgrade-cstor-pools-from-1.12.0-to-2.2.0.yaml
```

Comment:
Why do we need this line?

Reply:
No specific need. This line is created when `OPENEBS_OLD_VERSION` and `OPENEBS_NEW_VERSION` is rendered, hence when updating this document this line is not something that would need to be updated manually.